### DISCUSSION TITLE
Artisan data manipulation

### DISCUSSION THREAD

**Discussion by Microfiche62** [artisan-roaster-scope/artisan#1072] [Q&A]
Just started roasting recently with a Fresh Roast SR800, am am at roast 21 now. After a few roasts I decided to start using Artisan and go the Phidget and thermocouples route, because I am a data freak. However, there are a lot of moving pieces to keep track of during the roast and of course nothing other than the thermocouple temperature is being recorded automatically, so I am so far often missing things like hitting Charge, or moving the sliders I have set up for the fan and heat settings. I also keep a manual recording sheet and take a video of the roast at this point, so I can go back and see if I missed recording something etc.
So, looking for some help in fixing my Artisan roast data for posterity. In particular...

- How do I shift the time if I forgot to hit Charge, or started recording late - i.e. I need to add data at the beginning...
- How do I add/modify events like changing the fan to 9 or the heat to 4 from my slider input

Maybe it is with the event editor, but working with the entire dataset, fixing it, and re-importing it seems faster to me?
I am guessing the best way might be to export the data, fix it, and re-import it, but what format works best?
Also - how do I add events to each record - are they separated with a tab?
When I export to Artisan CSV, it doesn't look like the events are included, though they (sort of?) are in Excel, but not all of them?
Is there an export format that includes all those events as well?
I have done some searches, but not come up with anything that specifically discusses these topics...

Here is a sample where I hit charge 18 seconds after I started the roast...

![test](https://user-images.githubusercontent.com/74495468/211833952-154d364e-4180-468e-8a93-ac5b9a8f8524.png)

**Comment by poundy** [comment]
a) hit charge. When you get time, right click the actual charge point, and set "Charge" at that point.
b) Aren't these (air and burner) value changes already displayed on your example graph here? Not sure what you're trying to achieve here, as it seems to me that you've achieved your goal? Perhaps you want value labels displayed? Here, my burner values are shown on my resulting graph...
![image](https://user-images.githubusercontent.com/279966/211909301-a3d98e82-4507-4d8c-955d-ea38983a9cd9.png)

**Reply by poundy** [reply]
also, for a) you can change any milestone time that you need - FCS, drop etc. I'd also recommend that you get a workflow going where you do not charge without logging running, as I think what you're asking about adding new data in at the start of a roast isn't something anything will make "simple", but a workflow will make sure you don't have that issue...

**Reply by Microfiche62** [reply]
Thanks @poundy !

I have a few roasts that I was hoping to fix via the process of export / fix  / re-import as I don't think there is any way to "add" 18 seconds at the start of the roast profile (as in the profile I showed - the drop time was actually 11:00 minutes) to the start of the roast data _within the program_.
Also, there are a number of mistakes I made in the timing of the events - hit 30 seconds late/early or whatever. I was trying to figure out how to fix those within the profile itself, but it seems I would have to delete one entry, then look up to see what the ET and BT was at the correct time and add a new entry using that data? 
I just thought that seemed cumbersome and it might be easier to dump the data, fix it and re-import.

**Reply by Microfiche62** [reply]
> also, for a) you can change any milestone time that you need - FCS, drop etc. I'd also recommend that you get a workflow going where you do not charge without logging running, as I think what you're asking about adding new data in at the start of a roast isn't something anything will make "simple", but a workflow will make sure you don't have that issue...

Agreed - I am just starting this learning process so it will be smoother going forward. I was just hoping to salvage the data I do have.

**Reply by poundy** [reply]
If you hadn't started logging the data before the point shown on the graph, then no, there's no "simple" way to recreate that data into an existing roast.  I really don't think anyone here would have thought about this as a possible process, since it's so unusual, and would have so little relative value for anyone who is doing a large number of roasts. Personally, I'd say oh well, and move on and capture it better next time, next roast, next day etc etc... but I realise the desire for data... I can't really comment more on what options there might be to add data in your existing roasts, sorry.   You'd probably be best attaching a roast ALOG file here (copy one, add .txt as the end of the extension, and drag/drop into the reply) so we can see what data you have and you can describe what you see in the Roast Properties dialog box "Events" and "Data" tabs that you think is wrong and what you want to change.

**Reply by poundy** [reply]
I was just thinking as well, round-tripping the data out to CSV is possibly not the best, if someone can explain the ALOG file format in enough detail, I'd look at appending the missing bits in there rather than relying on export/import

**Reply by Microfiche62** [reply]
OK, that's cool. Appreciate the input!
I will keep looking deeper, but thought someone might have played with this before.

**Reply by MAKOMO** [reply]
Under menu Config >> Roast, Events tab you can edit the timing of events, values and descriptions in the event table. There you can also add and delete events and sort them by time.
